src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -534,7 +615,7 @@ public bool TryRemove(TKey key, out TValue value)
             }
 
             value = cacheItem.Value;
-            OnItemRemoved?.Invoke(cacheItem.Value);
+            OnItemMovedToCompactedList?.Invoke(cacheItem.Value);
```

Comment:
Do we need to add the cacheItem to _compactedItems?

Reply:
@GeoK If we move the cacheItem to the _compactedItems list, then eventually Dispose will be called.
We chose not to because this is a public API and could be called by the user, who may or may not want to dispose.

So, we just remove the item and set the signatureProvider.CryptoProviderCache to null.
Since EventBasedLRUCache doesn't know the type, we had to call a delegate defined in InMemoryCryptoProviderCache.